# Metagenomics

## Tools

### Shotgun metagenomics 
**Assemply**

 * [EULER]()
 * [Velvet]()
 * [SOAP]()
 * [ABySS]()
 * [MetaVelvet-SL]()
 * [Meta-IDBA]()
 * [Newbler (Rocher)]()
 * [MIRA]()
 * [Mapsembler]()
 * [ALLPATHS]()
 * [MetaORFA]()
 * [MetAMOS]()
 
**Binning**

 * [TETRA]()
 * [S-GSON]()
 * [PhylopythiaS]()
 * [TACOA]()
 * [PCAHIER]()
 * [ESOM]()
 * [ClaMS]()
 * [CARMA]()
 * [WGSQuikr]()
 * [SPHINX]()
 * [MetaPhyler]()
 * [SOrt-ITEMS]()
 * [PhymmBL]()
 * [MetaCluster]()
 
**Annotation** :white_check_mark:

 * [FASTX-Toolkit](http://hannonlab.cshl.edu/fastx_toolkit/)
 * [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
 * [SolexaQC](http://solexaqa.sourceforge.net/)
 * [Lucy 2](http://www.complex.iastate.edu/download/Lucy2/)
 * [DUST]() :shipit:
 * [Bowtie](http://bowtie-bio.sourceforge.net/index.shtml)
 * [MetaGeneMark](http://exon.gatech.edu/GeneMark/)
 * [LEfSe](https://bitbucket.org/biobakery/biobakery/wiki/lefse)
 * [TACOA](https://www.ncbi.nlm.nih.gov/pubmed/19210774)
 * [Metagene](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1636498/)
 * [CREST](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3527068/)
 * [Prodigal](http://prodigal.ornl.gov/)
 * [mOTU-LG](https://www.nature.com/nmeth/journal/v10/n12/full/nmeth.2693.html)
 * [Orphelia](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2703946/)
 * [Kraken](https://github.com/DerrickWood/kraken)
 * [FragGeneScan](https://github.com/wltrimbl/FGS)
 * [FragGeneScanPlus](https://github.com/hallamlab/FragGeneScanPlus)
 * [CTR]() :shipit:
 * [NBC](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3008645/)
 * [MyTaxa](https://www.ncbi.nlm.nih.gov/pubmed/24589583)
 * [RITA](http://kiwi.cs.dal.ca/Software/RITA)
 * [PILER-CR](http://drive5.com/pilercr/)
 * [tRNAscan](http://lowelab.ucsc.edu/tRNAscan-SE/)
 * [KEGG](http://www.genome.jp/kegg/)
 * [MetaCluster TA](http://i.cs.hku.hk/~alse/MetaCluster/index.html)
 * [SEED](http://www.theseed.org/wiki/Main_Page)
 * [eggNOG](https://bioinformatics.ca/links_directory/database/11646/eggnog) 
 * [ProViVE]() :shipit:
 * [COG/KOG](https://www.ncbi.nlm.nih.gov/COG/)
 * [PFAM](http://pfam.xfam.org/)
 * [TIGRFAM](http://www.jcvi.org/cgi-bin/tigrfams/index.cgi)
 * [HighSSR](https://goo.gl/2Sv3Q5)
 * [Blat](https://genome.ucsc.edu/FAQ/FAQblat.html)
 
**Analysis pipeline** :white_check_mark:
 
 * [IMG/MER](https://www.ncbi.nlm.nih.gov/pubmed/27738135)
 * [MG-RAST](https://www.ncbi.nlm.nih.gov/pubmed/26791506)
 * [MEGAN](http://ab.inf.uni-tuebingen.de/software/megan/)
 * [CAMERA](https://goo.gl/NMBSEY)
 * [Parallel-META](http://computationalbioenergy.org/parallel-meta.html)
 * [EBI Metagenomics](https://www.ebi.ac.uk/metagenomics/)
 * [METAREP](https://github.com/jcvi/METAREP)
 * [PHACCS](https://www.ncbi.nlm.nih.gov/pubmed/15743531)
 
### Marker genes metagenomics
**Standalone Sorfware**

**Analysis pipeline**

**Desosing** :white_check_mark:

 * [AmpliconNoise](https://github.com/fhcrc/ampliconnoise)
 * [DADA](https://sites.google.com/site/dadadenoiser/)
 * [JATAC](https://goo.gl/wp9HDH)
 * [UCHIME](http://drive5.com/usearch/manual/uchime_algo.html)
 * [Bellerophon](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bth226)
 * [CANGS](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2830946/)
 
## Articles 
 * [Sharpton - 2014 - An introduction to the analysis of shotgun metagenomic data](/papers/Sharpton-2014-An-introduction-to-the-analysis-of-shotgun-metagenomic-data.pdf)
 * [Oulas et al. - 2015 - Metagenomics Tools and Insights for Analyzing Next-Generation Sequencing Data Derived from Biodiversity Studies](./papers/Oulas-et-al.-2015-Metagenomics-Tools-and-Insights-for-Analyzing-Next-Generation-Sequencing-Data-Derived-from-Biodiversity-Studies.pdf)
 * [Mitchell et al. - 2016 - EBI metagenomics in 2016 - an expanding and evolving resource for the analysis and archiving of metagenomic dat](./papers/Mitchell-et-al.-2016-EBI-metagenomics-in-2016-an-expanding-and-evolving-resource-for-the-analysis-and-archiving-of-metagenomic-dat.pdf)
 * [Dudhagara et al. - 2015 - Web Resources for Metagenomics Studies](./papers/Dudhagara-et-al.-2015-Web-Resources-for-Metagenomics-Studies.pdf)
 * [Scheuch, Höper, Beer - 2015 - RIEMS a software pipeline for sensitive and comprehensive taxonomic classification of reads from metageno](./papers/Scheuch,-Höper,-Beer-2015-RIEMS-a-software-pipeline-for-sensitive-and-comprehensive-taxonomic-classification-of-reads-from-metageno.pdf)
